public/app/features/alerting/unified/plugins/PluginOriginBadge.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { useAsync } from 'react-use';
+
+import { Badge, Tooltip } from '@grafana/ui';
+
+import { getPluginSettings } from '../../../plugins/pluginSettings';
+
+interface PluginOriginBadgeProps {
+  pluginId: string;
+}
+
+export function PluginOriginBadge({ pluginId }: PluginOriginBadgeProps) {
+  const { value: pluginMeta } = useAsync(() => getPluginSettings(pluginId));
+
+  const logo = pluginMeta?.info.logos?.small;
+
+  const badgeIcon = logo ? (
+    <img src={logo} alt={pluginMeta?.name} style={{ width: '20px', height: '20px', borderRadius: '50%' }} />
+  ) : (
+    <Badge text={pluginId} color="orange" />
+  );
+
+  const tooltipContent = pluginMeta
+    ? `This rule is managed by the ${pluginMeta?.name} plugin`
+    : `This rule is managed by a plugin`;
+
+  return (
+    <Tooltip content={tooltipContent}>
+      <div>{badgeIcon}</div>
```

Comment:
From our chat before/for other reviewers: This is wrapped in a `div` as the `Badge` component doesn't forward a ref, so the tooltip wouldn't work in that case. Could forward a ref to it but wrapping in a div is easy enough 🤷